### DISCUSSION TITLE
How to use proxy with fetch API?

### DISCUSSION THREAD

**Discussion by ssslambda** [nodejs/undici#1105] [Q&A]
I'm rewriting legacy `request` code, it has `proxy` option, how i could achieve same behaviour with undici fetch API?

**Comment by ronag** [comment]
Maybe https://github.com/nodejs/undici/blob/main/examples/proxy-agent.js?

**Comment by jennyEckstein** [comment]
I attempted to build undici fetch API with proxy and I don't think I am successful. Please see repo to replicate [here](https://github.com/jennyEckstein/undici-fetch-proxy-example)
I took the original example [☝️](https://github.com/nodejs/undici/blob/main/examples/proxy-agent.js), run it with small modifications and then attempted to replicate the same results with fetch API.
Host in original example points to proxyUrl, and host in the fetch example points to serverUrl. So I can only assume something is not working correctly. 
Also, is there a better way to verify that proxy is being used?
<img width="1146" alt="Screen Shot 2022-01-03 at 4 50 38 PM" src="https://user-images.githubusercontent.com/7528575/147986271-5c7961ab-60b0-4358-90ee-a87c04007d42.png">

**Reply by ssslambda** [reply]
I also tried it and have same problem
